contracts/utils/ChainportLERC20.sol

```diff
@@ -0,0 +1,307 @@
+            lossless.beforeTransfer(_msgSender(), recipient, amount);
+        } 
+        _;
+    }
+
+    modifier lssTransferFrom(address sender, address recipient, uint256 amount) {
+        if (isLosslessOn) {
+            lossless.beforeTransferFrom(_msgSender(),sender, recipient, amount);
+        }
+        _;
+    }
+
+    modifier lssIncreaseAllowance(address spender, uint256 addedValue) {
+        if (isLosslessOn) {
+            lossless.beforeIncreaseAllowance(_msgSender(), spender, addedValue);
+        }
+        _;
+    }
+
+    modifier lssDecreaseAllowance(address spender, uint256 subtractedValue) {
+        if (isLosslessOn) {
+            lossless.beforeDecreaseAllowance(_msgSender(), spender, subtractedValue);
+        }
+        _;
+    }
+
+    modifier onlyRecoveryAdmin() {
+        require(_msgSender() == recoveryAdmin, "LERC20: Must be recovery admin");
+        _;
+    }
+
+    modifier onlyAdmin() {
+        require(_msgSender() == admin, "LERC20: Must be admin");
+        _;
+    }
+
+    modifier onlyChainportAdmin() {
+        require(_msgSender() == chainportAdmin, "LERC20: Must be Chainport Admin");
+        _;
+    }
+
+    modifier lssBurn(address account, uint256 amount) {
+        if (isLosslessOn) {
+            lossless.beforeBurn(account, amount);
+        } 
+        _;
+    }
+
+    modifier lssMint(address account, uint256 amount) {
+        if (isLosslessOn) {
+            lossless.beforeMint(account, amount);
+        } 
+        _;
+    }
+
+    // --- LOSSLESS management ---
+    function transferOutBlacklistedFunds(address[] calldata from) override external {
+        require(_msgSender() == address(lossless), "LERC20: Only lossless contract");
+
+        uint256 fromLength = from.length;
+        uint256 totalAmount = 0;
+        
+        for(uint256 i = 0; i < fromLength;) {
+            address fromAddress = from[i];
+            require(confirmedBlacklist[fromAddress], "LERC20: Blacklist not confirmed");
+            uint256 fromBalance = _balances[fromAddress];
+            _balances[fromAddress] = 0;
+            totalAmount += fromBalance;
+            emit Transfer(fromAddress, address(lossless), fromBalance);
+            unchecked{i++;}
+        }
+
+        _balances[address(lossless)] += totalAmount;
+    }
+
+    function setBlacklist(address[] calldata blacklist, bool value) external onlyChainportAdmin {
+        uint256 blacklistLenght = blacklist.length;
+        for(uint256 i = 0; i < blacklistLenght;) {
+            confirmedBlacklist[blacklist[i]] = value;
+            emit ConfirmBlacklisted(blacklist[i]);
+            unchecked{i++;}
+        }
+    }
+
+    function setLosslessAdmin(address newAdmin) override external onlyRecoveryAdmin {
+        require(newAdmin != admin, "LERC20: Cannot set same address");
+        emit NewAdmin(newAdmin);
+        admin = newAdmin;
+    }
+
+    function setChainportAdmin(address newAdmin) external onlyRecoveryAdmin {
```

Comment:
Let's make this use `onlyChainportAdmin` so that token admin does not have to deal with any of the chainport logic.

Reply:
Changed on new commit. Also added require on constructor in order to have the `chainportAdmin != address(0)` in order to not lock the functionality .